app/views/puppetclasses/_class_parameters.html.erb

```diff
@@ -1,6 +1,15 @@
 <% lookup_keys = overridable_lookup_keys(puppetclass, obj) %>
 <% lookup_keys.each_with_index do |lookup_key, index| %>
   <% lookup_value = lookup_value(obj, lookup_key) %>
+    <%
+       # LookupValue#match is generated automatically by Host#lookup_values_attributes=
+       # from the name, so don't present errors directly to the user.  Any validation
+       # errors should also be visible against the Host#name field.
+       if lookup_value.errors[:match].any?
```

Comment:
Perhaps checking for any .has-error inside that #params element would be enough (rather than tr or td specifically), there's probably nothing else that should have the has-error class in the tab.